src/gui/EmulatedUSBDevices/EmulatedUSBDeviceFrame.cpp

```diff
@@ -251,46 +240,14 @@ CreateSkylanderDialog::CreateSkylanderDialog(wxWindow* parent, uint8 slot)
 			return;
 
 		m_filePath = saveFileDialog.GetPath();
-
-		wxFileOutputStream output_stream(saveFileDialog.GetPath());
-		if (!output_stream.IsOk())
+		
+		if(!nsyshid::g_skyportal.CreateSkylander(_utf8ToPath(m_filePath.utf8_string()), skyId, skyVar))
 		{
-			wxMessageDialog saveError(this, "Error Creating Skylander File");
-			return;
+			wxMessageDialog errorMessage(this, "Failed to create file");
+			errorMessage.ShowModal();
+			this->EndModal(0);
```

Comment:
missing `return` after this? I dont think its intentional to call EndModal twice

Reply:
added this back in